plugins/modules/purefa_pg.py

```diff
@@ -491,203 +609,353 @@ def update_pgroup(module, array):
                 )
             changed = True
             if not module.check_mode:
-                try:
-                    array.set_pgroup(
-                        module.params["name"],
-                        targetlist=module.params["target"][0:4],
-                    )
-                except Exception:
-                    module.fail_json(
-                        msg="Changing targets for pgroup {0} failed.".format(
-                            module.params["name"]
+                for target in range(0, len(module.params["target"][0:4])):
+                    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                        res = array.post_protection_groups_targets(
+                            group_names=[module.params["name"]],
+                            context_names=[module.params["context"]],
+                            member_names=[FixedReference(name=target)],
+                        )
+                    else:
+                        res = array.post_protection_groups_targets(
+                            group_names=[module.params["name"]],
+                            member_names=[FixedReference(name=target)],
```

Comment:
member_names doesn't need FixedReference.
We can pass the list directly or use members as shown below.
members=[FixedReference(name=module.params["target"][target])]